python_modules/dagster/dagster/_daemon/controller.py

```diff
@@ -271,11 +285,23 @@ def check_daemon_heartbeats(self) -> None:
             )
 
     def check_workspace_freshness(self, last_workspace_update_time: float) -> float:
+        if self._skip_workspace_reload:
+            # Log the message only once
+            if not self._log_skip_message_logged:
+                self._logger.info("Skipping workspace reload due to debugging mode.")
+                self._log_skip_message_logged = True
+            return last_workspace_update_time
         nowish = get_current_timestamp()
         try:
-            if (nowish - last_workspace_update_time) > RELOAD_WORKSPACE_INTERVAL:
-                if self._grpc_server_registry:
-                    self._grpc_server_registry.clear_all_grpc_endpoints()
+            if self._grpc_server_registry and not self._daemon_reload_local_code_servers:
+                if not hasattr(self, '_log_grpc_skip_message_logged') or not self._log_grpc_skip_message_logged:
+                    self._logger.info("Not clearing GRPC endpoints. GRPC server is managed externally.")
+                    self._log_grpc_skip_message_logged = True
+            else:
+                if self._grpc_server_registry and not hasattr(self, '_log_grpc_clear_message_logged'):
+                    self._logger.info("Clearing GRPC endpoints due to workspace refresh.")
```

Comment:
it looks like the code that used to run here has been removed entirely - it is important that the old codepath still work